sbol_utilities/helper_functions.py

```diff
@@ -133,3 +133,25 @@ def has_plasmid_role(x):
         return is_plasmid(obj.instance_of.lookup())
     else:
         return False
+
+def is_composite(obj: sbol3.Component) -> bool:
+    """Check if an SBOL Component is a composite.
+
+    :param obj: design to be checked
+    :return: true if composite
+    """
+    def has_dna_type(o):
+        for t in o.types:
+            try:
+                regularized = tyto.SBO.get_uri_by_term(tyto.SBO.get_term_by_uri(t))
+                if regularized == tyto.SBO.deoxyribonucleic_acid:
+                    return True
+            except LookupError:
+                pass
+        return False
+
+    if isinstance(obj, sbol3.Component):  # Composite is SBOL component
+        if has_dna_type(obj):  # Composite has type SBO:DNA
+            if hasattr(obj, "generated_by") and len(obj.generated_by) != 0: # Composite has attribute prov:generated_by linking to prov:activity
```

Comment:
I think that checking for an SEP 054 assembly plan will need to have a more specific inspection of what sort of generation has occurred.

Reply:
I tried to check for entries in `generated_by` attribute whether they are assembly plan or not, but none of the files in the `test_files/` directory seems to contain a proper assembly plan. I tried checking wrt this point `Specifically, the Component for the composite part SHOULD have a prov:wasGeneratedBy propert whose value is a prov:Activity object with two values of its type property, one being sbol:design and the other being the new term http://sbols.org/v3#assemblyPlan`. can you please confirm whether this observation is correct or am I missing something.

Reply:
That is correct: all of the current test files were written before SEP 055 was drafted, and so don't contain an assembly plan in its form. You would want to create an example to check.